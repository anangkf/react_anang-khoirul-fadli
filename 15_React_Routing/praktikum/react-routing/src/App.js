import { PathRoutes } from './PathRoutes'

function App() {
  return (
    <>
      <PathRoutes />
    </>
  );
}

export default App;
