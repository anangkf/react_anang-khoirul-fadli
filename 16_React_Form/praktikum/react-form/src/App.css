*{
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: Roboto, sans-serif;
}

.container{
  display: flex;
}

.main{
  background-image: url('assets/img/bg-main.jpg');
  height: 100vh;
  width: 60%;
  background-size: cover;
  color: white;
}

.main header{
  height: 50%;
  text-align: center;
  padding: 3rem 1rem 0 1rem;
  background: linear-gradient(180deg, rgba(34,33,46,1) 0%, rgba(255,255,255,0) 100%);
  
}

form{
  margin: 16px 0 ;
}

.form-container{
  background-color: #fff;
  height: 100vh;
  width: 40%;
  padding: 10px 32px;
}

label{
  position: relative;
}

.form-container input:not([type='radio'], [type='file']), .form-container select{
  width: 100%;
  height: 2rem;
  position: relative;
  margin-bottom: 14px;
  padding: 0 12px;
  border: none;
  border-radius: 10px;
  background-color: #eedcf9;
  box-shadow: 0px 0px 66px -5px rgba(0,0,0,0.1);
}

.form-container input[type='radio']{
  margin: 2px 4px 14px 4px;
}

.form-container input[type='file']{
  margin: 2px 0 14px 0;
}

.form-container textarea{
  padding: 4px 12px;
  background-color: #eedcf9;
  width: 90%;
  height: 100px;
  border: none;
  border-radius: 10px;
  resize: none;
}

input:not([type='radio']):focus, textarea:focus{
  outline: 2px solid #0aa;
}

.required{
  color: #f22;
  font-weight: 700;
}

.error{
  position: absolute;
  left: 0;
  bottom: -24px;
  color: #f22;
  font-size: small;
}

button{
  padding: 4px 8px;
  background-color: transparent;
  color: #f22;
  font-weight: 500;
  font-size: medium;
  border: solid 1px #f22;
  border-radius: 20px;
  margin: 6px 8px;
  cursor: pointer;
}

button:hover{
  background-color: #f22;
  color: #FBFBFD;
}

button[type='submit']{
  color: #0c2;
  border: solid 1px #0c2;
}

button[type='submit']:hover{
  background-color: #0c2;
  color: #FBFBFD;
}

.confirm{
  color: #06d;
  border: solid 1px #06d;
  padding: 4px 12px;
}

.confirm:hover{
  background-color: #06d;
  color: #FBFBFD;
}

.alert{
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translateX(-50%) translateY(-50%);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 33%;
  height: 75vh;
  padding: 2.5rem 2rem;
  background-color: #FBFBFD;
  box-shadow: 14px 17px 48px -3px rgba(0,0,0,0.1);
  border-radius: 8px;
}

.alert header{
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 1.5rem;
}

.alert-logo{
  width: 25%;
  margin-bottom: .5rem;
}

.alert h3{
  text-align: center;
}

@media only screen and (max-width: 864px) {
  .container{
    display: flex;
    flex-direction: column;
  }

  .main{
    width: 100vw;
    height: 40vh;
  }

  .form-container{
    width: 100vw;
    height: 60vh;
  }

  .alert{
    width: 75%;
    height: 35%;
  }
}

@import url(
  'https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap'
);