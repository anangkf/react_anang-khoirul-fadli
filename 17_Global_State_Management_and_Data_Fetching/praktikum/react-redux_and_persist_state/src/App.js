import { SetupRouter } from './routes';

function App() {
  return (
    <>
      <SetupRouter />
    </>
  );
}

export default App;
