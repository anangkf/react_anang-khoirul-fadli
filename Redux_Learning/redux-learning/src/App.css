*, html, body{
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.navbar{
  background-color: #0bb;
  color: white;
  padding: 6px 24px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.count{
  position: relative;
  font-size: large;
  padding-right: 8px;
}

.count::after{
  content: attr(data-count);
  position: absolute;
  top: -10px;
  right: -10px;
  background-color: #ffc;
  color: #004;
  margin: 1px;
  text-align: center;
  font-size: 13px;
  width: 20px;
  height: 20px;
  border-radius: 40px;
}

.content{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 90vh;
  color: black;
}

.action-btn{
  margin-top: 32px;
  display: flex;
  justify-content: space-between;
  width: 240px;
}

.btn{
  background-color: #87a;
  padding: 8px 16px;
  border: none;
  border-radius: 18px;
  font-weight: bold;
  color: #fef;
  cursor: pointer;
}

.btn:hover{
  filter: brightness(.9);
}

.reset{
  background-color: #f33;
}