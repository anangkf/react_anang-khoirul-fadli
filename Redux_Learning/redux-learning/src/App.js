import './App.css';
import MainRoutes from './MainRoutes';
import Home from './pages/Home';

function App() {
  return (
    <MainRoutes />
    // <Home />
  );
}

export default App;
