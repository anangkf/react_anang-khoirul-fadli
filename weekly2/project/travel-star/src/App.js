import logo from './logo.svg';
import './App.css';
import Navbar from './components/Navbar';
import Hero from './components/Hero';
import ContactUs from './components/ContactUs';

function App() {
  return (
    <div className="">
      <Navbar />
      <main>
        <Hero />
        <ContactUs />
      </main>
      {/* <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header> */}
    </div>
  );
}

export default App;
