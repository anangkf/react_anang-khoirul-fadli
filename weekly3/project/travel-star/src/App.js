import logo from './logo.svg';
import './App.css';
import Navbar from './components/Navbar';
import Hero from './components/Hero';
import ContactUs from './components/ContactUs';
import AboutUs from './components/AboutUs';
import Footer from './components/Footer';

function App() {
  return (
    <div>
      <main>
        <Hero />
        <AboutUs />
        <ContactUs />
      </main>
      <Footer />
    </div>
  );
}

export default App;
