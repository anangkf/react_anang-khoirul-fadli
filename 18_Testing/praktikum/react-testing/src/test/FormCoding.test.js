import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../App';
import FormCoding from '../components/FormCoding'

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

describe('FormCoding', () =>{
    it('FormCoding SnapShots', () =>{
        const { container } = render(<FormCoding />);
		expect(container).toMatchSnapshot();
    })
    
    it('test each input to have expected values', () =>{
        render(<FormCoding />)
        // input nama
        const name = screen.getByRole('textbox', {name: /nama/i})
        fireEvent.input(name, 
            {target: {value: 'Anang'}}
            )
        expect(screen.getByLabelText(/nama lengkap/i)).toHaveValue('Anang')
        
        // input email
        const email = screen.getByRole('textbox', {name: /email/i})
        fireEvent.input(email, 
            {target: {value: 'mail@mail.com'}}
            )
        expect(screen.getByLabelText(/email/i)).toHaveValue('mail@mail.com')
        
        // input no handphone
        const phone = screen.getByRole('spinbutton', {name: /no handphone/i})
        fireEvent.input(phone, 
            {target: {value: '081293812931'}}
            )
        expect(screen.getByLabelText(/no handphone/i)).toHaveDisplayValue('081293812931')
        
        // input latar belakang pendidikan
        const background = screen.getByRole('radio', {name: /pendidikan/i})
        const IT = screen.getByLabelText(/IT/)
        const nonIt = screen.getByLabelText(/Non IT/)
        
        fireEvent.change(background)
        fireEvent.click(IT,
            { target: { value: "IT" } }
            );
        fireEvent.click(nonIt,
            { target: { value: "Non IT" } }
            );
        
        expect(screen.getByLabelText(/Latar Belakang Pendidikan:/)).toBeChecked();
        
        // input no handphone
        const selectedClass = screen.getByRole('combobox', {name: /kelas/i})
        userEvent.selectOptions(selectedClass, 'reactjs')
        expect(screen.getByLabelText(/Kelas Coding yang Dipilih:/)).toHaveValue("reactjs");
        
        // input file
        const fileTester = new File(["input"], "input.pdf", { name: /surat kesungguhan/i }, { type: "image/*,.pdf" });
        const fileInput = screen.getByLabelText(/Foto Surat Kesungguhan:/);
        // Check if file has not been choosen
		expect(fileInput.files.length).toBe(0);
		// Check if file has been choosen
		userEvent.upload(fileInput, fileTester);
		expect(fileInput.files.length).toBe(1);

        // input harapan
        const harapan = screen.getByRole("textbox", { name: /harapan/i })
        fireEvent.input(harapan,
            {target: {value: 'semoga lulus dengan nilai yang baik'} }
            );
        expect(screen.getByLabelText(/Harapan Untuk Coding Bootcamp Ini:/)).toHaveValue('semoga lulus dengan nilai yang baik');

    })
    // name field validation
    it('test name field if the value given are not valid', () =>{
        render(<FormCoding />)
        const name = screen.getByRole('textbox', {name: /nama/i})
        fireEvent.input(name, 
            {target: {value: 'Anang999'}}
            )
            expect(screen.getByText('Nama Lengkap Harus Berupa Huruf')).toBeInTheDocument()
        })
        
    // email field validation
    it('test email field if the value given are not valid', () =>{
        render(<FormCoding />)
        const email = screen.getByRole('textbox', {name: /email/i})
        fireEvent.input(email, 
            {target: {value: 'mailmailcom'}}
            )
        expect(screen.getByText('Email Tidak Sesuai')).toBeInTheDocument() 
    })
        
    // phone field validation
    it('test phone field if the value given are less than 9 or more than 14', () =>{
        render(<FormCoding />)
        const phone = screen.getByRole('spinbutton', {name: /no handphone/i})
        fireEvent.input(phone, 
            {target: {value: '08718721'}}
            )
        fireEvent.input(phone, 
            {target: {value: '087187218192812'}}
            )
        expect(screen.getByText('No Handphone Tidak Sesuai')).toBeInTheDocument() 
    })

    // check if user hit submit button
    it('test if user hit the submit button', () =>{
        render(<FormCoding />)
        const alertMock = jest.spyOn(window, "alert").mockImplementation();

        const name = screen.getByRole('textbox', {name: /nama/i})
        fireEvent.input(name, 
            {target: {value: 'Anang'}}
            )
        expect(screen.getByLabelText(/nama lengkap/i)).toHaveValue('Anang')

        const btnSubmit = screen.getByRole("button", { name: /submit/i })
        fireEvent.click(btnSubmit);
        expect(alertMock).not.toBe();
    })

    // check if user hit submit button with incomplete form
    it('test if user submitted incomplete form', () =>{
        render(<FormCoding />)

        const alertMock = jest.spyOn(window, "alert").mockImplementation();
        const name = screen.getByRole('textbox', {name: /nama/i})
        fireEvent.input(name, 
            {target: {value: ''}}
            )
        expect(screen.getByLabelText(/nama lengkap/i)).toHaveValue('')
        expect(screen.getByLabelText(/Nama Lengkap:/)).toBeRequired();
        
        // input email
        const email = screen.getByRole('textbox', {name: /email/i})
        fireEvent.input(email, 
            {target: {value: ''}}
            )
        expect(screen.getByLabelText(/email/i)).toHaveValue('')
        expect(screen.getByLabelText(/Email:/)).toBeRequired();
        
        // input no handphone
        const phone = screen.getByRole('spinbutton', {name: /no handphone/i})
        fireEvent.input(phone, 
            {target: {value: ''}}
            )
        expect(screen.getByLabelText(/no handphone/i)).toHaveDisplayValue('')
        expect(screen.getByLabelText(/No Handphone:/)).toBeRequired();
        
        // input latar belakang pendidikan
        const background = screen.getByRole('radio', {name: /pendidikan/i})
        const IT = screen.getByLabelText(/IT/)
        const nonIt = screen.getByLabelText(/Non IT/)
        fireEvent.change(background)
        fireEvent.click(IT,
            { target: { value: "" } }
            );
        fireEvent.click(nonIt,
            { target: { value: "" } }
            );
        expect(screen.getByLabelText(/Latar Belakang Pendidikan:/)).not.toBeChecked()
        expect(screen.getByLabelText(/Latar Belakang Pendidikan:/)).toBeRequired();
        
        // input no handphone
        const selectedClass = screen.getByRole('combobox', {name: /kelas/i})
        userEvent.selectOptions(selectedClass, 'reactjs')
        expect(screen.getByLabelText(/Kelas Coding yang Dipilih:/)).toHaveValue("reactjs");
        
        // input file
        const fileTester = new File(["input"], "input.pdf", { name: /surat kesungguhan/i }, { type: "image/*,.pdf" });
        const fileInput = screen.getByLabelText(/Foto Surat Kesungguhan:/);
        // Check if file has not been choosen
		expect(fileInput.files.length).toBe(0);
		// Check if file has been choosen
		userEvent.upload(fileInput, fileTester);
		expect(fileInput.files.length).toBe(1);

        // input harapan
        const harapan = screen.getByRole("textbox", { name: /harapan/i })
        fireEvent.input(harapan,
            {target: {value: 'semoga lulus dengan nilai yang baik'} }
            );
        expect(screen.getByLabelText(/Harapan Untuk Coding Bootcamp Ini:/)).toHaveValue('semoga lulus dengan nilai yang baik');

        const btnSubmit = screen.getByRole("button", { name: /submit/i })
        fireEvent.click(btnSubmit);
        expect(alertMock).not.toBe();
    })

    // check if user submitted invalid data
    it('check if user hit submit button when there are invalid data', () =>{
        render(<FormCoding />)
        const alertMock = jest.spyOn(window, "alert").mockImplementation();

        const name = screen.getByRole('textbox', {name: /nama/i})
        fireEvent.input(name, 
            {target: {value: 'Anang999'}}
            )
        expect(screen.getByText('Nama Lengkap Harus Berupa Huruf')).toBeInTheDocument()
            
        const email = screen.getByRole('textbox', {name: /email/i})
        fireEvent.input(email, 
            {target: {value: 'mailmailcom'}}
            )
        expect(screen.getByText('Email Tidak Sesuai')).toBeInTheDocument() 

        const phone = screen.getByRole('spinbutton', {name: /no handphone/i})
        fireEvent.input(phone, 
            {target: {value: '08718721'}}
            )
        fireEvent.input(phone, 
            {target: {value: '087187218192812'}}
            )
        expect(screen.getByText('No Handphone Tidak Sesuai')).toBeInTheDocument() 

        const btnSubmit = screen.getByRole("button", { name: /submit/i })
        fireEvent.click(btnSubmit);
        expect(alertMock).not.toBe();
    })
        
})