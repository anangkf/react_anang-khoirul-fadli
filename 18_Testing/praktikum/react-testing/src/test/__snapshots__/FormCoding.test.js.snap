// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCoding FormCoding SnapShots 1`] = `
<div>
  <h1
    style="text-align: center;"
  >
    Pendaftaran Peserta Coding Bootcamp
  </h1>
  <form
    class="centerForm"
  >
    <label>
      Nama Lengkap:
      <input
        class="input"
        name="nama"
        required=""
        type="text"
        value=""
      />
    </label>
    <label>
      Email: 
      <br />
      <input
        class="input"
        name="email"
        required=""
        type="email"
        value=""
      />
    </label>
    <label>
      No Handphone: 
      <br />
      <input
        name="noHandphone"
        required=""
        type="number"
        value=""
      />
    </label>
    <label>
      Latar Belakang Pendidikan: 
      <br />
      <input
        name="pendidikan"
        required=""
        type="radio"
        value="IT"
      />
      IT
      <input
        name="pendidikan"
        type="radio"
        value="Non IT"
      />
      Non IT
    </label>
    <br />
    <label>
      Kelas Coding yang Dipilih: 
      <br />
      <select
        name="kelas"
        required=""
        type="text"
      >
        <option
          disabled=""
          value=""
        >
          Pilih Salah Satu Program
        </option>
        <option
          value="golang"
        >
          Coding Backend with Golang
        </option>
        <option
          value="reactjs"
        >
          Coding Frontend with ReactJS
        </option>
        <option
          value="fullstack"
        >
          Fullstack Developer
        </option>
      </select>
    </label>
    <label>
      Foto Surat Kesungguhan:
      <input
        name="suratKesungguhan"
        required=""
        type="file"
      />
    </label>
    <label>
      Harapan Untuk Coding Bootcamp Ini: 
      <br />
      <textarea
        name="harapan"
        type="text"
      />
    </label>
    <ul />
    <br />
    <br />
    <input
      type="submit"
      value="Submit"
    />
    <button
      class="buttonReset"
    >
      Reset
    </button>
  </form>
</div>
`;
